agent/exec/containerd/adapter.go

```diff
@@ -0,0 +1,424 @@
+package containerd
+
+import (
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"os"
+	osexec "os/exec"
+	"path/filepath"
+	"strings"
+	"sync"
+	"syscall"
+
+	"github.com/Sirupsen/logrus"
+
+	"github.com/docker/containerd/api/services/execution"
+	"github.com/docker/containerd/api/types/container"
+
+	"github.com/docker/distribution/manifest"
+	"github.com/docker/distribution/manifest/schema1"
+	"github.com/docker/distribution/manifest/schema2"
+
+	"github.com/docker/swarmkit/agent/exec"
+	"github.com/docker/swarmkit/api"
+	"github.com/docker/swarmkit/api/naming"
+	"github.com/docker/swarmkit/log"
+
+	"github.com/opencontainers/runtime-spec/specs-go"
+
+	"github.com/pkg/errors"
+	"github.com/tonistiigi/fifo"
+	"golang.org/x/net/context"
+)
+
+// containerAdapter conducts remote operations for a container. All calls
+// are mostly naked calls to the client API, seeded with information from
+// containerConfig.
+type containerAdapter struct {
+	client execution.ExecutionServiceClient
+	//container *containerConfig
+	container    *api.ContainerSpec
+	task         *api.Task
+	secrets      exec.SecretGetter
+	bundle       string
+	contentStore string
+}
+
+func newContainerAdapter(client execution.ExecutionServiceClient, bundleDir, contentStore string, task *api.Task, secrets exec.SecretGetter) (*containerAdapter, error) {
+	ctx := context.Background()
+
+	// XXX steal from ../container/container.go?
+	//ctnr, err := newContainerConfig(task)
+	//if err != nil {
+	//	return nil, err
+	//}
+
+	container := task.Spec.GetContainer()
+	if container == nil {
+		return nil, exec.ErrRuntimeUnsupported
+	}
+
+	bundle := filepath.Join(bundleDir, task.ID)
+
+	log.G(ctx).Debugf("newContainerAdapter: ID=%s SVC=%s", task.ID, task.ServiceID)
```

Comment:
I think I can just kill this debug and do the `WithFields` thing elsewhere when there is actual logging to do.